### DISCUSSION TITLE
NSPanel with Humidity Sensor

### DISCUSSION THREAD

**Discussion by domerich** [Blackymas/NSPanel_HA_Blueprint#2360] [Ideas]
Hi guys. I was wondering if it was possible to remove the integrated temperature sensor of the NSPanel and replace it with the Bosch BME280 for better temperature accuracy and humidity reading?

I’m a noob but I think it should be possible as you have to open the thing to flash it anyway 

after flashing you could just solder the BME board to the RX TX points on the board 

![](https://blakadder.com/assets/images/nspanel/pinout.jpg)

https://esphome.io/cookbook/bme280_environment.html
https://tasmota.github.io/docs/BME280/

**Comment by edwardtfn** [comment]
Another alternative would be exploring the `H2` pins described by blakadder on his amazing [NSPanel teardown](https://blakadder.com/nspanel-teardown/).

> A peculiar detail is the unpopulated spot at the top labelled “H2”. I have my suspicions that it was supposed to be a humidity sensor, partly because of the H and partly because the firmware reports a static humidity value of 50 together with the temperature. Another case of an abandoned feature or a sign of things to come?

I bet those pins are connected to some ESP32 pin which would be a better alternative. But will require testing that wiring...

**Comment by edwardtfn** [comment]
Related:
- #2295
- #2316

**Reply by domerich** [reply]
> Related:
> 
> * [`Enhancement` Use pins from relais for other purpose #2295](https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/2295)
> * [Complete modification: BME680 with NSPanel: Accurate temp, humidity, gas sensor install inside the NSPanel, worked, want use bsec libary Bosch #2316](https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/2316)

This guy did a really nice job with the BME680 I might as well just make use of his efforts. Not sure if I have to drill a hole in the housing though

**Comment by edwardtfn** [comment]
I haven't tried that, but I don't see any reason for not working.
You probably will need some customization to get temperature in another way, but it shouldn't be that hard... I can try to help with that if you go this way.

**Comment by griffintel** [comment]
I managed to put a bme280 in mine and it's been working well.  I need to clean it up a bit, but pics can be found in this thread: https://reddit.com/r/homeassistant/comments/1dx4j16/sonoff_nspanel_esphome_free_gpio_pin_for/

**Comment by nils512** [comment]
Does someone knows if this also works with Tasmota?
